bindings/python/tests/bindings/test_processors.py

```diff
@@ -73,3 +79,78 @@ def test_processing(self, roberta_files):
         output = tokenizer.encode("My name is John")
         assert output.tokens == ["ĠMy", "Ġname", "Ġis", "ĠJohn"]
         assert output.offsets == [(0, 2), (3, 7), (8, 10), (11, 15)]
+
+
+class TestTemplateProcessing:
+    def get_bert(self):
+        return TemplateProcessing(
+            seq_a=["[CLS]", "$0", "[SEP]"],
+            seq_b=["$1", "[SEP]"],
+            special_tokens=[("[CLS]", 1), ("[SEP]", 0)],
+        )
+
+    def get_roberta(self):
+        return TemplateProcessing(
+            seq_a="<s> $0 </s>", seq_b="</s> $0 </s>", special_tokens=[("<s>", 0), ("</s>", 1)],
+        )
+
+    def get_t5_squad(self):
+        # >>> from transformers import AutoTokenizer
+        # >>> tok = AutoTokenizer.from_pretrained("t5-small")
+        # >>> tok.tokenize("question: ")
+        # ['▁question', ':']
+        # >>> tok.tokenize("context: ")
+        # ['▁context', ':']
+        # >>> tok.encode("context: ")
+        # [2625, 10]
+        # >>> tok.encode("question: ")
+        # [822, 10]
+
+        return TemplateProcessing(
+            seq_a=["Q", "$0"],
+            seq_b=["C", "$1"],
+            special_tokens=[
+                {
+                    "id": "Q",
+                    "ids": [2625, 10],
+                    "type_ids": [None, None],
+                    "tokens": ["_question", ":"],
+                },
+                {
+                    "id": "C",
+                    "ids": [822, 10],
+                    "type_ids": [None, None],
+                    "tokens": ["_context", ":"],
+                },
+            ],
+        )
+
+    def test_instantiate(self):
+        bert = self.get_bert()
+        assert bert is not None
+        assert isinstance(bert, PostProcessor)
+        assert isinstance(bert, TemplateProcessing)
+        assert isinstance(pickle.loads(pickle.dumps(bert)), TemplateProcessing)
+
+    def test_bert_parity(self):
+        tokenizer = Tokenizer(BPE())
+        tokenizer.add_special_tokens(["[SEP]", "[CLS]"])
+        tokenizer.add_tokens(["my", "name", "is", "john", "pair"])
+        tokenizer.post_processor = BertProcessing(("[SEP]", 0), ("[CLS]", 1))
+
+        original = tokenizer.encode("my name", "pair")
+
+        tokenizer.post_processor = self.get_bert()
+        template = tokenizer.encode("my name", "pair")
+        assert original.ids == template.ids
```

Comment:
I am a bit confused about ("[CLS]", 0) and the `.add_special_tokens`.

I tried to create a similar tokenizer, but it seems doing `tokenizer.add_special_tokens(["[SEP]", "[CLS]"])` add "[SEP]" with id=0. However I defined my PostProcessor with `special_tokens=[("[CLS]", 0), ("[SEP]", 1)]`. I will end up with [0, 2, 3, 1, 6, 1]. I think it should be incorrect, or something or something should yell at me. However I'm not sure how all this is supposed to work. Maybe the two "[SEP]" are different ?

Reply:
:+1: Opened #406 to track this.